lib/ddtrace/pin.rb

```diff
@@ -0,0 +1,62 @@
+# \Datadog global namespace that includes all tracing functionality for Tracer and Span classes.
+module Datadog
+  # A \Pin (a.k.a Patch INfo) is a small class which is used to
+  # set tracing metadata on a particular traced object.
+  # This is useful if you wanted to, say, trace two different
+  # database clusters.
+  class Pin
+    def self.get_from(obj)
+      return nil unless obj.respond_to? :datadog_pin
+      obj.datadog_pin
+    end
+
+    attr_accessor :service
+    attr_accessor :app
+    attr_accessor :tags
+    attr_accessor :app_type
+    attr_accessor :name
+    attr_accessor :tracer
+
+    def initialize(service, params = {})
```

Comment:
Nit: you could use named params with defaults to be _slightly_ more self documenting and idiomatic, if we need ruby 2 anyway.